Add support for TILEDB_VAR_NUM sized attributes in filter expressions and move to latest muparserx

nalinigans opened the pull request.

mlathara commented on core/src/expressions/expression.cc:
```diff
@@ -141,35 +160,130 @@ T get_value(const void *buffer, const uint64_t buffer_index) {
-            break;
-          }
+          default:
+            assign_fixed_cell_values(attribute_id, buffers, buffer_sizes, j, positions[i]);
```
fixed and single versions don't need `buffer_sizes`

mlathara reviewed (commented):
back to you @nalinigans with some questions

nalinigans commented on core/src/expressions/expression.cc:
```diff
@@ -141,35 +160,130 @@ T get_value(const void *buffer, const uint64_t buffer_index) {
+      std::pair<size_t, size_t> info = get_var_cell_info<int>(buffers, buffer_sizes, buffer_index, position);
+      mup::Value x_array(mup::int_type(info.first), mup::int_type(info.second));
+      for (auto i=0u; i<info.second; i++) {
+        x_array.At(i) = get_single_cell_value(attribute_type, buffers, buffer_sizes, buffer_index, info.first+i);
```
Yes, good catch. Don't have a unit test yet for this path :-( , the only path that was getting tested was for strings that was handling it as buffer_index+1.

nalinigans commented:
@mlathara, I think I have addressed your review comments. Also, wrote a whole bunch of catch2 unit tests with templates to test out all the different paths. Think it is ready for a review again. Thanks.

mlathara reviewed (commented):
looks good. do you think it's worth adding a test exercising var-sized attributes where the entries are actually variable sized?

nalinigans commented:
> looks good. do you think it's worth adding a test exercising var-sized attributes where the entries are actually variable sized?

Sure, I have added a test exercising var-sized attributes and variable offsets. We will need to discuss adding custom operations to figure out operations like empty, exists, length to make the expressions useful. Should we complete that before merging this PR?

mlathara reviewed (approved):
I think merge this in now. This can be useful even without those operations.
